text/unstable_to_title_case.ts

```diff
@@ -0,0 +1,83 @@
+// Copyright 2018-2025 the Deno authors. MIT license.
+// This module is browser compatible.
+import { resolveOptions, titleCaseWord } from "./_title_case_util.ts";
+import type { BaseTitleCaseOptions } from "./_title_case_util.ts";
+export type { BaseTitleCaseOptions };
+
+/**
+ * A function that filters words in the string. If a word returns `false` from this function, it will not be title-cased.
+ * @param value The word to be filtered.
+ * @param index The index of the word in the array.
+ * @param array The array of words.
+ * @returns `true` if the word should be title-cased, `false` otherwise.
+ */
+export type WordFilter = (
+  value: Intl.SegmentData,
+  index: number,
+  array: Intl.SegmentData[],
+) => boolean;
+
+/** Options for {@linkcode toTitleCase} */
+export interface TitleCaseOptions extends BaseTitleCaseOptions {
+  /**
+   * A function that filters words in the string. If a word returns `false`
+   * from this function, it will not be title-cased.
+   *
+   * Alternatively, you can pass an array of strings to filter out specific
+   * words.
+   *
+   * @default {() => true} (no filtering)
+   */
+  filter?: WordFilter | readonly string[];
+}
+
+/**
+ * Converts a string into Title Case.
+ *
+ * @experimental **UNSTABLE**: New API, yet to be vetted.
+ *
+ * @param input The string that is going to be converted into Title Case
+ * @param options Optional settings to customize the conversion
+ * @returns The string as Title Case
+ *
+ * @example Usage
+ * ```ts
+ * import { toTitleCase } from "@std/text/unstable-to-title-case";
+ * import { assertEquals } from "@std/assert/equals";
+ *
+ * assertEquals(toTitleCase("deno is awesome"), "Deno Is Awesome");
+ * ```
+ */
+export function toTitleCase(input: string, options?: TitleCaseOptions): string {
+  const opts = resolveOptions(options);
+  // [3.13.2 Default Case Conversion `toTitlecase`](https://www.unicode.org/versions/Unicode16.0.0/core-spec/chapter-3/#G34078)
+  // toTitlecase(X): Find the word boundaries in X according to Unicode Standard Annex #29, “Unicode Text Segmentation.”
+  const segments = [...opts.segmenter.segment(input)];
```

Comment:
I wonder this splitting of words matches the user's expectation.

Other `toFooCase` APIs use `splitToWords` utility, and it splits camel case, pascal case, kebab case, constant case, etc into parts, but this doesn't do it. Is that difference intentional?

ex.
```js
toKebabCase("HelloWorld") // => hello-world
toSnakeCase("HelloWorld") // => hello_world
toConstantCase("HelloWorld") // => HELLO_WORLD
toTitleCase("HelloWorld") // => Helloworld // <- this is one word in this case
```

Reply:
Ok. Probably this parsing of the input makes sense for `toTitleCase` and `toSentenceCase`. Can you describe how these 2 APIs handle the input in API docs, with some notes about the difference with other `toXCase` APIs?